app/src/main/java/com/jvoye/tasky/auth/presentation/login/LoginScreen.kt

```diff
@@ -59,16 +59,20 @@ fun LoginScreenRoot(
 
     LoginScreen(
         state = state,
-        onAction = viewModel::onAction,
-        onSignUpClick = onSignUpClick
+        onAction = { action ->
+            when(action) {
+                is LoginAction.OnSignUpClick -> onSignUpClick
```

Comment:
For the navigation issue, you’re passing a function reference instead of calling it. `onSignUpClick` (no parentheses) doesn’t invoke anything.😅 . I already tried this fix and it worked.
``` kotlin
when (action) {
    is LoginAction.OnSignUpClick -> onSignUpClick() // add ()
    else -> { /* ... */ }
}
```